infor-design/enterprise-ng
# Reverting back to the dual package.json and adding helper scripts

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
For ease of adoption, we reverted back to the old dual package.json (from PR https://github.com/infor-design/enterprise-ng/pull/92) ways so that the npm package would have a specific file hierarchy. See the issue for more details.

**Related github/jira issue (required)**:
Issue #107 (jira: `SOHO-8134`)
Issue #108 

**Steps necessary to review your pull request (required)**:

## New Badges
1. Go to https://github.com/infor-design/enterprise-ng/tree/npm-postinstall-hook
1. Verify there are 3 new badges at the top of the README:
    - License
    - NPM package
    - build

## Test the file hierarchy:
1. `npm run call-prepublishOnly && npm pack publish/`
1. Go to the zip file and extract it to examine the contents match the following screenshot, plus the `CHANGELOG.md` and `LICENSE`
1. Verify it looks like the screenshot and remember to delete the `.zip` and the `package/` folder it creates before moving on)
     ![npm pack publish files](https://user-images.githubusercontent.com/742319/41177834-770673a4-6b33-11e8-929b-3e8cecacb768.png)

## Test appending the version date script
1. `npm run version-bump:dev`
1. Verify that the `publish/package.json` *version* property has the date `.YYYYMMDD` appended.
1. If it worked, do `git checkout publish/package.json` to remove the change

## Test the update-enterprise script
1. `npm run update-enterprise`
1. Select a tag in the cli (it'll prompt you)
1. Verify the `./package.json` "ids-enterprise" version is at the [latest for its tag](https://www.npmjs.com/package/ids-enterprise)
1. Verify the `./publish/package.json` "ids-enterprise" version is the same as the `./package.json`
1. Verify `/src/soho/icon/` has the following files:
    - svg-extended.html
    - svg-patterns.html
    - svg.html
    - svg-empty.html
1. If it worked, do `git reset --hard` to remove the changes

**Closing issues**
Fixes #108 
Fixes #107
